java-checks/src/main/java/org/sonar/java/checks/MethodParametersOrderCheck.java

```diff
@@ -68,13 +68,14 @@ public void visitNode(Tree tree) {
     }
     ParametersList formalParameterList = parametersByMethod.computeIfAbsent(methodInvTree.symbol(), m -> new ParametersList(methodDeclaration));
     List<IdentifierTree> argumentsList = methodInvTree.arguments().stream().map(this::argumentToIdentifier).collect(Collectors.toList());
-    if (matchingNames(formalParameterList, argumentsList)
-      && matchingTypesWrongOrder(formalParameterList, argumentsList)) {
-      List<JavaFileScannerContext.Location> flow = methodDeclaration.parameters().stream().map(param -> new JavaFileScannerContext.Location("Formal Parameters", param))
-        .collect(Collectors.toList());
-      reportIssue(methodInvTree.arguments(), "Parameters to " + methodInvTree.symbol().name() + " have the same names but not the same order as the method arguments.",
-        flow, null);
-
+    if (matchingNames(formalParameterList, argumentsList)) {
+      List<VariableTree> matchingTypesWrongOrder = matchingTypesWrongOrder(formalParameterList, argumentsList);
+      if (!matchingTypesWrongOrder.isEmpty()) {
+        List<JavaFileScannerContext.Location> flow = matchingTypesWrongOrder.stream().map(param -> new JavaFileScannerContext.Location("Formal Parameters", param))
```

Comment:
As this was also causing some trouble, I would still go for the change you previously did: replacing the strange `Formal parameters` by `Misplaced parameter`.